plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoPageSourceProvider.java

```diff
@@ -55,6 +57,32 @@ public ConnectorPageSource createPageSource(
             handles.add((MongoColumnHandle) handle);
         }
 
-        return new MongoPageSource(mongoSession, tableHandle, handles.build());
+        if (dynamicFilter.isAwaitable()) {
+            dynamicFilter.isBlocked().join();
```

Comment:
PageSource should not wait for dynamic filter. Probably you want to alter `MongoSplitManager` to test if DF is completed and use it during split scheduling. Also, there is DynamicFIlterConfig with parameter how long we want to await for DF to complete.